### PR TITLE
dialects: add riscv Mul et co

### PR DESCRIPTION
**Pull request by muke101**
Added some unimplemented riscv ops - mul, mulh, div, divu etc. All tests pass.

### CODE REVIEW

**Comment by muke101** [comment]
@superlopuh

File: tests/filecheck/dialects/riscv/riscv_assembly_emission.mlir [hunk 1]
```diff
@@ -7,8 +7,6 @@
   // CHECK-NEXT: li j1, 5
   %2 = "riscv.add"(%0, %1) : (!riscv.reg<j1>, !riscv.reg<zero>) -> !riscv.reg<j2>
   // CHECK-NEXT: add j2, zero, j1
-  %mv = "riscv.mv"(%0) : (!riscv.reg<zero>) -> !riscv.reg<j2>
```
Comment (superlopuh) [inline tests/filecheck/dialects/riscv/riscv_assembly_emission.mlir:10]: please add back :)

File: tests/filecheck/dialects/riscv/riscv_ops.mlir [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // RUN: xdsl-opt %s | xdsl-opt | filecheck %s
 "builtin.module"() ({
-  %0 = "riscv.get_register"() : () -> !riscv.reg<>
-  %1 = "riscv.get_register"() : () -> !riscv.reg<>
+  %0 = "test.op"() : () -> !riscv.reg<>
```
Comment (superlopuh) [inline tests/filecheck/dialects/riscv/riscv_ops.mlir:3]: please change back :)

File: xdsl/dialects/riscv.py [hunk 3]
```diff
@@ -470,6 +440,31 @@ def __init__(
         )
 
 
+class RdRsOperation(IRDLOperation, RISCVOp, ABC):
```
Comment (superlopuh) [inline xdsl/dialects/riscv.py:443]: might as well reduce diff here as well

**Review by superlopuh — CHANGES_REQUESTED** [review]
please rebase your changes on main

**Comment by superlopuh** [comment]
Can you run black, please? Should just be `black .`

**Review by superlopuh — APPROVED** [review]
beautiful

**Pull request closed without merging**